### ISSUE TITLE
inconsistent disabling of predictable interface names causes networking.service failure

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#6420]
PR #6403 disabled predictable network interfaces during image setup (`*.setup`), but *not* during bootstrap. Thus now on the Debian images `networking.service` fails to start on boot. `/etc/network/interfaces` still has
```
# The primary network interface
auto ens2
iface ens2 inet dhcp
```
(which was set up by virt-builder presumably), but the unit fails with
```
* networking.service - Raise network interfaces
   Loaded: loaded (/lib/systemd/system/networking.service; enabled; vendor preset: enabled)
  Drop-In: /run/systemd/generator/networking.service.d
           `-50-insserv.conf-$network.conf
   Active: failed (Result: exit-code) since Tue 2017-05-02 02:28:08 EDT; 2min 53s ago

May 02 02:28:08 unassigned-hostname ifup[998]: Cannot find device "ens2"
May 02 02:28:08 unassigned-hostname dhclient[1043]: Error getting hardware address for "ens2": No such device
May 02 02:28:08 unassigned-hostname ifup[998]: Error getting hardware address for "ens2": No such device
```
`eth0` is now being brought up by NetworkManager asynchronously during boot. This is fine for our tests as they don't expect to be online, but nevertheless is an inconsistency.

If we need to keep this (I'm not a fan of disabling the interface naming I must say), it needs to be done in the bootstrap scripts too.

I noticed this while debugging something else and it's not a blocker, so I don't address it right now.

**Comment by martinpitt** [comment]
Fedora images also still have a `/etc/sysconfig/network-scripts/ifcfg-ens2` with `ONBOOT="yes"`, but the boot sequence doesn't seem to complain if the device is absent. `ifcfg-eth0` also exists, and this is being read/executed with NetworkManager, so these images work.

This bug could potentially explain the random test failure that I [now see](https://github.com/cockpit-project/cockpit/pull/6407#issuecomment-298506750) on the ubuntu-1604 image:
```
# testUnmanaged (check_networking_unmanaged.TestNetworking)
Unexpected journal message 'audit: type=1400 audit(1493674171.748:12): apparmor="DENIED" operation="connect" info="Failed name lookup - disconnected path" error=-13 profile="/usr/lib/NetworkManager/nm-dhcp-helper" name="run/NetworkManager/private-dhcp" pid=6794 comm="nm-dhcp-helper" requested_mask="wr" denied_mask="wr" fsuid=0 ouid=0'
```
The AppArmor violation itself is of course an Ubuntu bug, but this change is likely to have triggered it as bringing up the primary networkg now moved from ifupdown to NM.

**Comment by martinpitt** [comment]
I can reproduce this locally with adding a `sit()` and running `systemctl restart NetworkManager`. The DHCP helper gets launched for eth0:
```
root      7508  0.0  0.3  20500  3956 ?        S    03:06   0:00 /sbin/dhclient -d -q -sf /usr/lib/NetworkManager/nm-dhcp-helper -pf /var/run/dhclient-eth0.pid -lf /var/lib/NetworkManager/dhclient-b95913e0-5002-4b2c-8c9c-1c7fa458afb0-eth0.lease -cf /var/lib/NetworkManager/dhclient-eth0.conf eth0
```
which pretty much confirms that this change has been the trigger.

Of course the actual bug is in AppArmor or the profiles somewhere. But I cannot reproduce this on a standard Ubuntu xenial (cloud) image with manual commands (looks like this is rather timing sensitive), so I haven't yet filed a downstream bug report.

**Comment by martinpitt** [comment]
Turns out this isn't actually part of what virt-builder does, but `/etc/network/interfaces` is already like that in the pristine ~/.cache/virt-builder/ubuntu-16.04.x86_64.1 image; I uncompressed, kpartx'ed and loop-mounted that to confirm. On the other hand, the debian-8.x86_64.1 image already has an /etc/network/interfaces entry for eth0, as in Jessie the udev names weren't yet being used by default. So I propose to just sed that straight in debian.setup, together with adding the boot parameter for net.ifnames, to keep the hack in one place.

This doesn't really affect the Fedora or RHEL images, as explained above.
